VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift

```diff
@@ -19,40 +19,40 @@ extension NSCalendar {
         components.month = month
         components.day = day
         components.hour = 12
-        return dateFromComponents(components)
+        return dateFromComponents(components)!
     }
     
     func dateForTomorrowRelativeToToday(today: NSDate) -> NSDate {
         let tomorrowComponents = NSDateComponents()
         tomorrowComponents.day = 1
-        return dateByAddingComponents(tomorrowComponents, toDate: today, options: nil)
+        return dateByAddingComponents(tomorrowComponents, toDate: today, options: [])!
     }
     
     func dateForEndOfWeekWithDate(date: NSDate) -> NSDate {
         let daysRemainingThisWeek = daysRemainingInWeekWithDate(date)
         let remainingDaysComponent = NSDateComponents()
         remainingDaysComponent.day = daysRemainingThisWeek
-        return dateByAddingComponents(remainingDaysComponent, toDate: date, options: nil)
+        return dateByAddingComponents(remainingDaysComponent, toDate: date, options: [])!
     }
     
     func dateForBeginningOfDay(date: NSDate) -> NSDate {
-        let newComponent = components((NSCalendarUnit.YearCalendarUnit | NSCalendarUnit.CalendarUnitMonth | NSCalendarUnit.CalendarUnitDay), fromDate: date)
+        let newComponent = components(([.Year, .Month, .Day]), fromDate: date)
         let newDate = dateFromComponents(newComponent)
-        return newDate
+        return newDate!
     }
     
     func dateForEndOfDay(date: NSDate) -> NSDate {
         let components = NSDateComponents()
         components.day = 1
         let toDate = dateForBeginningOfDay(date)
-        let nextDay = dateByAddingComponents(components, toDate: toDate, options: nil)
-        let endDay = nextDay.dateByAddingTimeInterval(-1)
+        let nextDay = dateByAddingComponents(components, toDate: toDate, options: [])!
+//        let endDay = nextDay.dateByAddingTimeInterval(-1)
```

Comment:
This might be a bug, that we are returning `nextDay` instead of `endDay`. I guess we should add a test for that :-)